src/data/bucket/symbol_bucket.js

```diff
@@ -679,10 +684,10 @@ class SymbolBucket implements Bucket {
             cos = Math.cos(angle);
 
         symbolInstanceIndexes.sort((aIndex, bIndex) => {
-            const a = this.symbolInstances[aIndex];
-            const b = this.symbolInstances[bIndex];
-            const aRotated = Math.round(sin * a.anchor.x + cos * a.anchor.y) | 0;
-            const bRotated = Math.round(sin * b.anchor.x + cos * b.anchor.y) | 0;
+            const a = this.symbolInstances.get(aIndex);
+            const b = this.symbolInstances.get(bIndex);
```

Comment:
Let's make sure sorting doesn't get more expensive here — perhaps worth trying to build `rotated` and `featureIndices` arrays first and then only do `(rotated[aIndex] - rotated[bIndex]) || (featureIndices[bIndex] - featureIndices[aIndex])` in the comparison function.

Reply:
Good idea, I'll do that. :+1: